src/modules/transaction/hooks/useTransactionFee/useTransactionFee.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+import { useCommandSchema } from '@network/hooks';
+import { useAuth } from '@auth/hooks/queries';
+import { computeFee, getParamsSchema } from './utils';
+/**
+ * 
+ * @param {object} data
+ * @param {boolean} data.isValid Whether the transaction is valid or not. TxComposer defines this
+ * @param {object} data.wallet The sender account info as returned by useDeprecatedAccount
+ * @param {object[]} data.priorities An array of object as { value: number, title: string, selected: boolean }
+ * @param {object} data.transaction Transaction object as Lisk Element expects without fee
+ * @returns {object} The fee object with a total value, and a component value as an array of fees
+ * that contribute in the total value
+ */
+export const useTransactionFee = ({
+  isValid,
+  wallet,
+  priorities,
+  transaction,
+}) => {
+  const [fee, setFee] = useState(0);
+  const { data: auth } = useAuth({ config: { params: { address: wallet?.summary?.address } } });
+  const { moduleCommandSchemas } = useCommandSchema();
+  const schema = getParamsSchema(transaction, moduleCommandSchemas);
+
+  useEffect(() => {
+    setFee(computeFee(transaction, schema, auth, priorities, isValid));
+  }, [
+    priorities,
+  ]);
+
+  return {
+    total: fee,
+    components: [
+      { value: fee, type: 'bytesFee' },
+    ],
+  };
+};
```

Comment:
so useEffect dependcies is missing evrey single value inside the useEffect is a dependence other than var from global scope, there is no need for an setState witch is coused an extra re-render
```suggestion
const fees = usememo( () => {
const fee = computeFee(transaction, schema, auth, priorities, isValid);
return {
  totalFee: fee,
  fees: [{ fee, type: 'bytesFee' }],
}
, [transaction, schema, auth, priorities, isValid])
 return fees
```